### ISSUE TITLE
kv: acceptance/gossip/peerings flaked

### ISSUE THREAD

**Issue by msbutler** [cockroachdb/cockroach#96091]
See flake in CI [here](https://teamcity.cockroachdb.com/viewLog.html?buildId=8469572&problemId=11989).

This issue depends on resolving #74714

Jira issue: CRDB-23944

**Comment by tbg** [comment]
```
* ERROR: ERROR: server startup failed: cockroach server exited with error: result is ambiguous: error=grpc: error reading from server: read tcp 127.0.0.1:56366->127.0.0.1:26263: read: connection reset by peer [code 14/Unavailable] [exhausted]
* SQLSTATE: 40003
*
ERROR: server startup failed: cockroach server exited with error: result is ambiguous: error=grpc: error reading from server: read tcp 127.0.0.1:56366->127.0.0.1:26263: read: connection reset by peer [code 14/Unavailable] [exhausted]
SQLSTATE: 40003
Failed running "start"
```

and I came here to report another one:

https://teamcity.cockroachdb.com/viewLog.html?buildId=8648958&buildTypeId=Cockroach_BazelEssentialCi

```
I230209 14:47:30.888468 342 kv/kvserver/store_remove_replica.go:150 â‹® [T1,n1,s1,r58/1:â€¹/{Table/57-Max}â€º,raft] 108  removing replica r58/1
I230209 14:47:31.041859 1 1@cli/start.go:900 â‹® [T1,n1] 109  initiating graceful shutdown of server
I230209 14:47:31.042094 1 1@cli/start.go:961 â‹® [T1,n1] 110  too early to drain; used hard shutdown instead
E230209 14:47:31.043987 1 1@cli/clierror/check.go:35 â‹® [T1] 111  server startup failed: cockroach server exited with error: result is ambiguous: error=ba: â€¹Put [/Table/46/1/1/0,/Min), EndTxn(parallel commit) [/Table/46/1/1/0], [txn: eca4807c], [can-forward-ts]â€º RPC error: grpc: â€¹node waiting for init; /cockroach.roachpb.Internal/Batch not availableâ€º [code 14/Unavailable] [exhausted]
```

In other words, while we're sleeping here, 

https://github.com/cockroachdb/cockroach/blob/f2bbad960cd3c9e0a46633374050fb32bb648137/pkg/cmd/roachtest/tests/gossip.go#L303-L306

the node crashes during startup. (There isn't a `monitor` running in this test that would expose this more directly).

**Comment by aliher1911** [comment]
Looks like we are hitting #74714 here with ambiguous error. It is not even node n3 that was just restarted that failed to start, but previously restarted n1 failed.
We can add monitor here to make more transparent, but it won't fix the underlying problem of failing nodes.

**Comment by rafiss** [comment]
Flaked for me at https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_BazelEssentialCi/8898308?showRootCauses=false&expandBuildChangesSection=true&expandBuildTestsSection=true&expandBuildProblemsSection=true

The [test history](https://teamcity.cockroachdb.com/test/-6408532727161146414?currentProjectId=Cockroach_Ci_Tests&expandTestHistoryChartSection=true) shows that 31 of the last 1000 runs have failed (3% failure rate).

**Comment by msbutler** [comment]
Flaked again. 
https://teamcity.cockroachdb.com/viewLog.html?buildId=8991367&buildTypeId=Cockroach_BazelEssentialCi

Going to skip

**Comment by erikgrinaker** [comment]
Thanks. I'm removing GA-blocker since this problem exists in 22.1 as well, but we're actively working on it.

**Comment by erikgrinaker** [comment]
I think this would actually have been resolved by #100213, going to unskip.

**Comment by knz** [comment]
just flaked again. https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_Tests_LocalRoachtest/9407543?showRootCauses=false&expandBuildChangesSection=true&expandBuildTestsSection=true

**Comment by knz** [comment]
skipping again in #100483

**Comment by erikgrinaker** [comment]
The failure in that build is different than the previous failure modes, in that it now straight-up can't connect to n1, which may or may not be expected. Will look into it later.

**Comment by ajwerner** [comment]
https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_BazelEssentialCi/9411491?showRootCauses=false&expandBuildChangesSection=true&expandBuildProblemsSection=true&expandBuildTestsSection=true

**Comment by erikgrinaker** [comment]
The last couple of failures are a test artifact: the test begins restarting nodes before other nodes have completed startup, and those other nodes are in the middle of running RPCs on the restarted node.

```
E230403 20:28:31.633739 1 1@cli/clierror/check.go:35 ⋮ [-] 88 +server startup failed: cockroach server exited with error: ‹migration-job-find-already-completed›: grpc: ‹error reading from server: read tcp 127.0.0.1:39384->127.0.0.1:26257: read: connection reset by peer› [code 14/Unavailable]
```

:26257 is node 1, which was restarted at 20:28:31:

```
20:28:31 gossip.go:301: 1: restarting node 1
20:28:31 cluster.go:683: test status: stopping nodes :1
local: stopping and waiting
```

This will incidentally have been fixed by #74714 that merged yesterday, but this test should really be waiting for nodes to complete startup before killing other nodes.
